effekt/shared/src/main/scala/effekt/source/Tree.scala

```diff
@@ -544,7 +538,40 @@ object Named {
     case AnyPattern  => symbols.ValueParam
   }
 
-  type ResolvedReferences[T <: References] = T match {
+  import effekt.context.Annotations
+
+  /**
+   * Definitions resolve to exactly one symbol, which is the one they introduce
+   */
+  extension [T <: Resolvable](t: T & Definition) {
+    def symbol(using C: Context): SymbolFor[T] =
+      symbolOption getOrElse {
+        C.panic(s"Internal Compiler Error: Cannot find symbol for ${t.id}")
+      }
+
+    def symbolOption(using C: Context): Option[SymbolFor[T]] =
+      C.annotationOption(Annotations.Symbol, t.id).map(_.asInstanceOf[SymbolFor[T]])
+  }
+
+  /**
+   * `ValDef`s resolve to potentially multiple symbols.
+   */
+  extension (t: ValDef) {
+    def boundSymbols(using C: Context): List[effekt.symbols.Binder.ValBinder] =
+      // the symbol of a ValueParam is a symbols.ValueParam, not a ValBinder!
+      t.binders.map { (b: ValueParam) => b.symbol; ??? }
```

Comment:
@phischu I purposefully crash here. Can we look into this together? We need to decide how to represent them on the symbols level.

Reply:
We could conflate the two. I just want to mention that having separate symbols for different constructs makes generating better error messages in Intelligence easier, because we can simply match on the type and specialize the error.

Reply:
I also don't like the name `boundSymbols`, but since `symbols` is a package name, calling it like this will crash everything....